src/main/kotlin/domain/usecase/task/CreateTaskUseCase.kt

```diff
@@ -6,7 +6,7 @@ import org.example.domain.repository.TaskRepository
 class CreateTaskUseCase(
     private val taskRepository: TaskRepository
 ) {
-    fun createTask(task: Task): Result<Task> {
+    fun createTask(task: Task): Task {
```

Comment:
we can make it in one line 
```suggestion
        fun createTask(task: Task): Task = taskRepository.createTask(task)
```